app/controllers/admin/assets_controller.rb

```diff
@@ -13,28 +13,30 @@ def index
     def unspam
       @asset.ham!
       @asset.update_attribute :is_spam, false
-      flash.notice = "Track was made public"
-      redirect_back(fallback_location: root_path)
     end
 
     def spam
```

Comment:
We probably need to `soft_delete` on spam and `restore` on unspam like we do with users and have appropriate behavior in the frontend to match — this could be a followup with a migration to soft delete the existing tracks marked as spam if you don't want to do it here.

Reply:
We have a clear separation of `Spam/Unspam` from `Delete/Restore` in users.
I would prefer to keep those boundaries clear here as well. We might not want to delete spam tracks right away (for research purposes :). Especially if we have an automatic permanent deletion of an asset after a certain time. 

We could add a job that would `DeleteAllSpamAssets` or something very precise like that. Thoughts?

Reply:
> We have a clear separation of Spam/Unspam from Delete/Restore in users.

Marking a user as a spam always soft deletes it: https://github.com/sudara/alonetone/blob/master/app/controllers/admin/users_controller.rb#L19-L24

I'm not sure we'd want to change that deviate from "spam means deleted" mental model without a good reason. The implication on the UI side is "users are spammed and deleted with one click but tracks require 2 clicks" — Is the reason technical? (i.e. that we'd have to switch out a lot of `not_spam` scopes for asset?)

Reply:
Moved to #672